### ISSUE TITLE
Research heading character count guidance

### ISSUE THREAD

**Issue by sarah-sch** [uswds/uswds-site#2343]
### Purpose
To research guidance for appropriate header lengths.

### Background
At Dev Sync on Oct. 27, we discussed the serif heading being used as a default for the Prose component. The team discussed the appropriate length of the headings and subheadings.

### Definition of done
We have researched heading length guidance, aligned on appropriate guidance for our community, and decided how this guidance should be reflected on the. website.

**Comment by finekatie** [comment]
CS/UX Sync 11/15. An issue was on the agenda about character count but passed over for time being. Just noting here. Not sure if it's the same thing.

**Comment by annepetersen** [comment]
Nope, this is about the recommended character count of headings, while the agenda item was about character count in form entry.

**Comment by finekatie** [comment]
Sarah and Katie did some initial research. H1s have standard character count limits of 50-60 characters for SEO to begin with ([see SEM Rush](https://www.semrush.com/blog/h1-tag/)), so brevity is necessary. As far as H2-H6, there is no specific limit [guidance on WCAG](https://www.w3.org/WAI/standards-guidelines/act/rules/b49b2e/proposed/). The comment we see featured is this: Headings do not need to be lengthy. A word, or even a single character, may be sufficient.

**Comment by finekatie** [comment]
Leaving note here for @sarah-sch to possibly bring to DevSync on Thurs 12/14

**Comment by finekatie** [comment]
Hi @annepetersen - I think this warrants a revisit. Would want to make sure whatever guidance we add is consistent wherever we're talking about headers and limits. Unless this was just supposed to be specific to the Prose component, per the original description. @sarah-sch , your thoughts?

**Comment by sarah-sch** [comment]
I am adding to the CS/UX sync agenda for next week. @finekatie and I found guidance encouraging content creators to be concise with headings, but we didn't find anything around a specific character count (other than for SEO purposes when dealing with H1s). @annepetersen - I think you had mentioned considering cognitive issues (e.g., dyslexia) as we continue to research this issue. Anything else to consider as we conduct a little background research ahead of the sync next week?

**Comment by sarah-sch** [comment]
Thanks, Anne! That direction is helpful.

**Comment by sarah-sch** [comment]
@annepetersen and @finekatie - I don't know if we will have time to discuss at CS/UX/a11y sync today, but I did spend more time on this question of heading length limit and am sharing my findings below. cc @jaclinec, @amycole501, @alex-hull, @thisisdano

### Takeaways

- Descriptive, concise headings can support readers with dyslexia.
- If we need to provide a character limit for headings, we might start with the 60-70 character optimal sentence length supported by the British Dyslexia Association. Consider setting our heading limit at the low end (60) or a little below that, assuming that optimal heading length would be shorter than optimal sentence length.
- It could be worthwhile to compare the British Dyslexia Association’s style guide with USWDS guidance and consider updating guidance where we disagree with BDA.

### The research on line length

- My quick review of the literature generally supports USWDS’s recommendation of 45-90 characters per line / around 66 characters per line. It seems pretty consistent with best practices that I see elsewhere, although 90 cpl might be pushing it. For example, the [Baymard Institute](https://baymard.com/blog/line-length-readability) suggests that 50-75 cpm is optimal.
- Generally, people with dyslexia [seem to prefer relatively short line lengths](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0071161), like the ones you might find on mobile devices.
- Optimal line length may vary according to what you are measuring (reading speed vs. comprehension vs. retention), who you are measuring (experts vs. novices), and other factors, like whether someone has dyslexia. [This interesting post by Mary Dyson](https://www.reading.ac.uk/typography/staff/dr-mary-dyson), a psychology researcher currently in the Dept. of Typography and Graphic Communication at the University of Reading suggests that there isn’t a perfect scientific consensus on optimal line length yet! 

### Recommendations specific to dyslexia

- As stated above, people with dyslexia seem to prefer a line length shorter than what might be preferred by people without dyslexia.
- There seem to be a lot of other text considerations that help folks with dyslexia, many of which come up in the [Audioeye blog post you shared above](https://www.audioeye.com/post/designing-for-dyslexia/). Some that jump out to me as key include: sufficient spacing between characters and allowing users to customize line spacing, fonts, font sizes, etc.
- [This style guide from the British Dyslexia Association](https://www.bdadyslexia.org.uk/advice/employers/creating-a-dyslexia-friendly-workplace/dyslexia-friendly-style-guide) seems to fall in line with the research I came across in my (admittedly non-comprehensive) review of the literature. On headings, the style guide says:
Use headings to provide structure
Heading font size should be 20% larger than normal text
Be generous with spacing around headings

### Heading character count

- I didn’t find any recommendations specific to heading character count (again, beyond the 50-60 H1 recommendation for SEO), but in general what seems key for dyslexia is finding the right balance between being both descriptive and concise. 
- The British Dyslexia Association guide suggests 60-70 characters as an optimal length for sentences, so I think we would be safe to suggest that as a limit for headings.
- I like this quote from Jeff Curtis in the Audioeye post that Anne shared: “Use short, descriptive headlines: For people with language or memory impairments, descriptive headlines can help process pages faster and navigate to the content they want to read.”

**Comment by brunerae** [comment]
@sarah-sch / @finekatie what are the next steps for this Issue? Is it still a priority?

**Comment by finekatie** [comment]
Discussed today. Anne would like to revisit this. Not super pressing, but has priority energy. Ashley moved to next sprint.
